var/spack/repos/builtin/packages/gtkplus/package.py

```diff
@@ -45,10 +45,15 @@ class Gtkplus(AutotoolsPackage):
     depends_on("pango~X", when='~X')
     depends_on("pango+X", when='+X')
     depends_on('gobject-introspection', when='+X')
+    depends_on('shared-mime-info')
 
     patch('no-demos.patch')
 
     def patch(self):
         # remove disable deprecated flag.
         filter_file(r'CFLAGS="-DGDK_PIXBUF_DISABLE_DEPRECATED $CFLAGS"',
                     '', 'configure', string=True)
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        spack_env.prepend_path("XDG_DATA_DIRS",
+                               self.prefix.share)
```

Comment:
I have set `XDG_DATA_DIRS` at run-time in all packages that define it for builds. It will be interesting to see what breaks as a result, because when `XDG_DATA_DIRS` is defined, it replaces all other data directories used by gtk applications. For example, it will reveal packages that have undocumented dependencies on data provided by the operating system. We may need to undo these latest changes if they cause too many difficulties.